examples/07_join_aggregation.py

```diff
@@ -0,0 +1,438 @@
+from sklearn.preprocessing import OneHotEncoder
+
+col_trans = make_column_transformer(
+    ("passthrough", ["userId", "movieId", "rating"]),
+    (OneHotEncoder(sparse_output=False), ["day_name"]),
+    remainder="drop",
+    verbose_feature_names_out=False,
+)
+
+train = col_trans.fit_transform(train)
+train = pd.DataFrame(train, columns=col_trans.get_feature_names_out())
+train.head()
+
+
+# ***Join aggregator on `movieId` and `userId`***
+
+# In[11]:
+
+
+from skrub import JoinAggregator
+
+cols_day_name = list(set(train.columns) - set(["userId", "movieId"]))
+
+join_agg_rating = JoinAggregator(
+    tables=[
+        (train, "userId", cols_day_name),
+        (train, "movieId", cols_day_name),
+    ],
+    main_key=["userId", "movieId"],
+    suffixes=["_user", "_movie"],
+    agg_ops=["sum", "mean"],
+)
+train = join_agg_rating.fit_transform(train[["userId", "movieId", "rating"]])
+
+cols_to_drop = [
+    col
+    for col in train.columns
+    if ("mean" in col and "day_name" in col) or ("rating_mean" in col)
+]
+train = train.drop(cols_to_drop, axis=1)
+train
+
+
+# ### Encoding movies
+
+# In[12]:
+
+
+movies = fetch_movielens(dataset_id="movies")
+movies = movies.X
+movies
+
+
+# In[13]:
+
+
+movies["genres"] = movies["genres"].str.replace("|", " ")
+all_genres = movies["genres"].str.split().explode().unique()[:-3]
+all_genres
+
+
+# In[14]:
+
+
+from sklearn.feature_extraction.text import CountVectorizer
+from skrub import MinHashEncoder
+
+vectorizer = CountVectorizer(
+    vocabulary=all_genres,
+    ngram_range=(1, 1),
+    lowercase=False,
+)
+
+n_components = 10
+min_hash_encoder = MinHashEncoder(n_components=n_components)
+
+movie_transformer = make_column_transformer(
+    ("passthrough", ["movieId"]),
+    (min_hash_encoder, ["title"]),
+    (vectorizer, "genres"),
+    sparse_threshold=0,
+)
+movie_embedding = movie_transformer.fit_transform(movies)
+
+embedding_cols = [f"title_x{idx}" for idx in range(n_components)] + all_genres.tolist()
+movie_embedding = pd.DataFrame(movie_embedding, columns=["movieId", *embedding_cols])
+movie_embedding
+
+
+# ## Encoding users
+
+# In[15]:
+
+
+train_user_movie_embedding = train[["userId", "movieId"]].merge(
+    movie_embedding, on="movieId", how="left"
+)
+train_user_movie_embedding.shape
+
+
+# ## Join Aggregate user and movie embeddings!
+
+# In[16]:
+
+
+join_agg_embedding = JoinAggregator(
+    tables=[
+        (movie_embedding, "movieId", embedding_cols),
+        (train_user_movie_embedding, "userId", embedding_cols),
+    ],
+    main_key=["movieId", "userId"],
+    agg_ops=["mean"],
+    suffixes=["_movie", "_user"],
+)
+train_embedding = join_agg_embedding.fit_transform(train)
+train_embedding
+
+
+# ## Putting everything together!
+
+# In[17]:
+
+
+def get_year_and_day_name(ratings):
+    timestamps = pd.to_datetime(ratings["timestamp"], unit="s")
+    day_names = timestamps.dt.day_name()
+    return pd.get_dummies(day_names)
+
+
+pipe_ohe_days = make_pipeline(
```

Comment:
Could we get away with a DatetimeEncoder here? (if not, could we improve the DatetimeEncoder so that it works here?)

Trying to simplify this notebook as much as I can